include/devicetree.h

```diff
@@ -145,6 +145,30 @@
  */
 #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
 
+
+/**
+ * @brief Get node name from its name or label.
+ *
+ * Example, in deficetree fragment:
+ *
+ *     serial0: serial@40002000 {
+ *             status = "okay";
+ *             current-speed = <115200>;
+ *             ...
+ *     };
+ *
+ * Node name is serial@40002000, which in definitions takes from
+ * serial_40002000, and serial0 is label, which takes form serial_0.
+ * Invoking:
+ *
+ *     DT_NODE_NAME(serial_40002000)
+ *     DT_NODE_NAME(serial_0)
```

Comment:
I can see this working on node labels (the second example with serial_0), because node labels are unique. But path components are not globally unique, so I don't see how the first example can work in general without running into problems with duplicates.

Reply:
> 
> 
> I can see this working on node labels (the second example with serial_0), because node labels are unique. But path components are not globally unique, so I don't see how the first example can work in general without running into problems with duplicates.

Ah, I missed that.